src/main/java/com/mentoring/ecommerce/application/port/in/SaveProductUseCase.java

```diff
@@ -0,0 +1,8 @@
+package com.mentoring.ecommerce.application.port.in;
+
+import com.mentoring.ecommerce.domain.Product;
+
+public interface SaveProductUseCase {
```

Comment:
Both `Controller` and `UseCase` from the dummy project use annotations (`@WebAdapter` and `@UseCase`).

Why was that implemented? Aren't these annotations redundant as, for instance, `@RestController` has already defined `@Target` and `@Retention`?

Should I add them to our project?

Reply:
Should I use `@UseCase` as well?

If not, should I declare use case classes as at least `@Component`?

Reply:
I recommend the use of `@UseCase.`